src/node_buffer.cc

```diff
@@ -741,6 +749,46 @@ void SlowByteLengthUtf8(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(args[0].As<String>()->Utf8Length(env->isolate()));
 }
 
+#if NODE_HAS_SIMD_NEON
+uint32_t FastByteLengthUtf8(Local<Value> receiver,
+                            const v8::FastOneByteString& source) {
+  uint8_t* data =
+      const_cast<uint8_t*>(reinterpret_cast<const uint8_t*>(source.data));
+  auto length = source.length;
+
+  uint32_t result{0};
+  const int lanes = sizeof(uint8x16_t);
+  uint8_t remaining = length % lanes;
+  const auto* simd_end = data + (length / lanes) * lanes;
```

Comment:
```suggestion
  const auto* simd_end = data + length;
```
Couldn't this be simplified like this?

Reply:
You must have that `data<simd_end` implies that `data+lanes <= length`. See code below.

Reply:
So no, I don't think that the proposed change is correct. Not that it almost surely does not get compiled into a multiplication and division. It is probably compiled to something like `data + length&15`.